dot/rpc/http.go

```diff
@@ -119,10 +130,19 @@ func (h *HTTPServer) Start() error {
 		}
 	}()
 
+	// init and start block received listener routine
+	if h.serverConfig.BlockAPI != nil {
+		h.serverConfig.BlockAddedReceiver = make(chan *types.Block)
+		h.serverConfig.BlockAddedReceiverDone = make(chan bool)
```

Comment:
this can be changed to `make(chan struct{})` and then in `Stop()` you can do `close(BlockAddedReceiverDone)` to notify the BlockState that it's done. it'll receive empty struct and unblock on `doneNotifying` 

it's a nitpick but I think it's usually what's used in go if you want a channel that only notifies

Reply:
Ok, updated.  I like the idea of applying patterns consistently (although I don't always do it), so thanks for pointing this out.